src/lib/decodeEvent.ts

```diff
@@ -44,6 +45,8 @@ export function decodeEvent(
   iface: Interface,
   eventProps: EventProps,
 ): DecodedEventResult {
+  if (!isHex(eventProps.data)) eventProps.data = '0x' + eventProps.data;
```

Comment:
This is kinda user input validation/adjustment. I think we should handle it earlier. As soon as we get it. So in the component. The data passed to logic functions should be already correct and the input from the user should be fixed as fast as possible.